### PR TITLE
[Merged by Bors] - feat: simp lemma for coercions in TopCat

### PR DESCRIPTION
**Pull request by kim-em**
This lemma doesn't have much effect right now in Mathlib, although it makes an example I'm writing for a tutorial go more smoothly. I could add that example in `test/` or even `doc/` if preferred, but maybe easiest not to?

I'm also looking at the simp lemmas around coercions and `of` in other bundled categories, and hoping to make them more uniform (e.g. `coe_of` sometimes is and sometimes isn't a simp lemma, mostly according to the vagaries of the porting process, I think!). The equivalent of this lemma in some other bundled categories seems helpful, but I'm still investigating.

### CODE REVIEW

File: Mathlib/CategoryTheory/Extensive.lean [hunk 1]
```diff
@@ -287,6 +287,7 @@ instance types.finitaryExtensive : FinitaryExtensive (Type u) := by
 
 section TopCat
 
+set_option maxHeartbeats 250000 in
```
Comment (mattrobball) [inline Mathlib/CategoryTheory/Extensive.lean:290]: This is `continuity`'s fault.

**Comment by mattrobball** [comment]
bors delegate+

File: Mathlib/CategoryTheory/Extensive.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/CategoryTheory/Extensive.lean]: @erdOne's new proof avoids it beautifully!

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**